### PR TITLE
Feature/base select

### PR DESCRIPTION
**Pull request by antek-ka**
### PR #7

**Feature/BaseSelect**
- Rebuild BaseSelect as custom component
- Minor fixes & updates

BaseSelect

BaseSelect_open

### CODE REVIEW

File: src/components/global/BaseSelect.vue [hunk 1]
```diff
@@ -10,71 +10,127 @@ export default defineComponent({
   },
   props: {
     selectOptions: {
-      type: Array,
-      default() {
-        return [];
-      }
```
Comment (szymonlicau) [inline src/components/global/BaseSelect.vue]: Why the removal of the default? 🤔

Comment (antek-ka) [reply src/components/global/BaseSelect.vue]: I was tinkering with a different approach of displaying the `currentOption` which the default value was somehow breaking and forgot to change it.. Should have left a comment for myself!

Comment (szymonlicau) [reply src/components/global/BaseSelect.vue]: I'd definitely try to leave the default value though for the array of options 😄

Comment (antek-ka) [reply src/components/global/BaseSelect.vue]: Yes - totally - in this solution it doesn't break anything, I'll change it immediately!

Comment (antek-ka) [reply src/components/global/BaseSelect.vue]: Done :smile:

**Review by szymonlicau — APPROVED** [review]

**Pull request merged**